Reduce copy-pasted code in PlotContainer with a decorator

neutrinoceros opened the pull request:
## PR Summary

I noticed that a handful of PlotContainer methods started with very similar lines to handle the `field="all"` case. I added a decorator to specifically do that and applied it to this functions instead.

I also extend its usage to `PlotContainer.set_background_color` (which is my original motivation for this PR)

question to reviewers: this is my first attempt at a decorator for class methods, do you think it would fit better _inside_ the class that uses it ?

neutrinoceros commented:
The change now contains a bugfix in `PlotContainer.set_minorticks`

neutrinoceros commented:
Will testing this for all affected callbacks, I (re)discovered a bug in set_colorbar_minorticks, which doesn't work at all (even on the master branch). See #2543

munkm commented:
Also, I know I asked for a few nitpicky things but I hope you know that your enhancements are very much appreciated!!!

neutrinoceros commented on yt/visualization/plot_container.py:
```diff
@@ -102,6 +102,25 @@ def newfunc(*args, **kwargs):
         return args[0]
     return newfunc
 
+def accept_allflag(f):
```
Thank you ! I was actually _hoping_ someone would come up with a better name during review :D

neutrinoceros commented:
> I know you added a bugfix in this PR too, and I'm wondering if you could pull it out into a separate PR? If we do a patch release we'll want to include that, but probably not the contents of this entire PR.

@munkm sure, sounds reasonable

neutrinoceros commented:
I've isolated the bugfix into its own PR, which the present own is now rebased on (I'll be able to rebase on master again after  #2547 is merged).

I've changed the decorator's name following your suggestion. I'll try writing some tests for it tomorrow.

neutrinoceros commented:
So actually I’ve been scratching my head over this but I really don’t know how to properly write a unit test for a _decorator_ :/
This one in particular is pretty entrenched in the PlotContainer class, so I'm a bit lost as to how to do this.

neutrinoceros commented:
update: rebased on master again

munkm commented:
Hmmm yeah we can't really test the decorator easily without using a plot object. What I'd suggest is passing an object with several fields and then doing a callback that uses that decorator. Rather than doing an answer test, do assertions to make sure that the attribute associated with annotation is the same for all of the fields based on what you've set.

matthewturk commented:
That sounds like a very good plan.

On Fri, Apr 17, 2020 at 2:39 PM Madicken Munk <notifications@github.com>
wrote:

> Hmmm yeah we can't really test the decorator easily without using a plot
> object. What I'd suggest is passing an object with several fields and then
> doing a callback that uses that decorator. Rather than doing an answer
> test, do assertions to make sure that the attribute associated with
> annotation is the same for all of the fields based on what you've set.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/yt-project/yt/pull/2525#issuecomment-615429978>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAVXO5ACMX2AZFUWOPU5VTRNCWAXANCNFSM4MEUYNCA>
> .
>

neutrinoceros commented:
Here's my first shot at this test ! Rather than targeting a real callback (prone to change and potentially decoupled from the decorator in the future), I attempted mocking a new callback.
